src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUntypedObject.cs

```diff
@@ -297,37 +297,64 @@ public static FormulaValue Float_UO(EvalVisitor runner, EvalVisitorContext conte
             return GetTypeMismatchError(irContext, BuiltinFunctionsCore.Value_UO.Name, DType.Number.GetKindString(), impl);
         }
 
-        public static FormulaValue Text_UO(EvalVisitor runner, EvalVisitorContext context, IRContext irContext, UntypedObjectValue[] args)
-        {
-            var impl = args[0].Impl;
-
+        public static FormulaValue Text_UO(EvalVisitor runner, EvalVisitorContext context, IRContext irContext, FormulaValue[] args)
+        {
+            if ((args[0] is UntypedObjectValue uo && uo.Type == FormulaType.Blank) || args[0] is BlankValue)
+            {
+                if (args.Length == 1)
+                {
+                    // As a special case, blank propagates for the single argument only
+                    return args[0];

```

Comment:
>return args[0]; [](http://example.com/codeflow?start=20&length=15)

We should return a blank value - which won't be the case on `Text(ParseJSON("null"))` - it will be returning an UntypedObjectValue